src/hypothesis/searchstrategy/numbers.py

```diff
@@ -474,6 +474,10 @@ class FullRangeFloats(FloatStrategy):
         (u'negative_probability', u'subnormal_probability')
     )
 
+    def __init__(self, allow_nan=True):
```

Comment:
It took me a moment to figure out why there wasn't a separate allow_infinity value here. Am I right in thinking that allow_nan=False will also disable infinity, because they're both represented with maximum exponent?